votingpool/withdrawal.go

```diff
@@ -0,0 +1,939 @@
+	return s.outputs
+}
+
+// Sigs returns a map of ntxids to signature lists for every input in the tx
+// with that ntxid.
+func (s *WithdrawalStatus) Sigs() map[Ntxid]TxSigs {
+	return s.sigs
+}
+
+// Fees returns the total amount of network fees included in all transactions
+// generated as part of a withdrawal.
+func (s *WithdrawalStatus) Fees() btcutil.Amount {
+	return s.fees
+}
+
+// NextInputAddr returns the votingpool address that should be used as the
+// startAddress of subsequent withdrawals.
+func (s *WithdrawalStatus) NextInputAddr() WithdrawalAddress {
+	return s.nextInputAddr
+}
+
+// NextChangeAddr returns the votingpool address that should be used as the
+// changeStart of subsequent withdrawals.
+func (s *WithdrawalStatus) NextChangeAddr() ChangeAddress {
+	return s.nextChangeAddr
+}
+
+// String makes OutputRequest satisfy the Stringer interface.
+func (r OutputRequest) String() string {
+	return fmt.Sprintf("OutputRequest %s to send %v to %s", r.outBailmentID(), r.Amount, r.Address)
+}
+
+func (r OutputRequest) outBailmentID() OutBailmentID {
+	return OutBailmentID(fmt.Sprintf("%s:%d", r.Server, r.Transaction))
+}
+
+// outBailmentIDHash returns a byte slice which is used when sorting
+// OutputRequests.
+func (r OutputRequest) outBailmentIDHash() []byte {
+	if r.cachedHash != nil {
+		return r.cachedHash
+	}
+	str := fmt.Sprintf("%s%d", r.Server, r.Transaction)
+	hasher := fastsha256.New()
+	// hasher.Write() always returns nil as the error, so it's safe to ignore it here.
+	_, _ = hasher.Write([]byte(str))
+	id := hasher.Sum(nil)
+	r.cachedHash = id
+	return id
+}
+
+func (o *WithdrawalOutput) String() string {
+	return fmt.Sprintf("WithdrawalOutput for %s", o.request)
+}
+
+func (o *WithdrawalOutput) addOutpoint(outpoint OutBailmentOutpoint) {
+	o.outpoints = append(o.outpoints, outpoint)
+}
+
+// Status returns the status of this WithdrawalOutput.
+func (o *WithdrawalOutput) Status() string {
+	return o.status.String()
+}
+
+// Address returns the string representation of this WithdrawalOutput's address.
+func (o *WithdrawalOutput) Address() string {
+	return o.request.Address.String()
+}
+
+// Outpoints returns a slice containing the OutBailmentOutpoints created to
+// fulfill this output.
+func (o *WithdrawalOutput) Outpoints() []OutBailmentOutpoint {
+	return o.outpoints
+}
+
+// Amount returns the amount (in satoshis) in this OutBailmentOutpoint.
+func (o OutBailmentOutpoint) Amount() btcutil.Amount {
+	return o.amount
+}
+
+// withdrawal holds all the state needed for Pool.Withdrawal() to do its job.
+type withdrawal struct {
+	roundID         uint32
+	status          *WithdrawalStatus
+	transactions    []*withdrawalTx
+	pendingRequests []OutputRequest
+	eligibleInputs  []Credit
+	current         *withdrawalTx
+	// newWithdrawalTx is a member of the structure so it can be replaced for
+	// testing purposes.
+	newWithdrawalTx func() *withdrawalTx
```

Comment:
Does this actually need to be unique for every withdrawal?  I'd think this should be package level instead of keeping an additional member for every withdrawal instance.